frontend/src/metabase/writeback/components/ActionCreator/ActionCreator.tsx

```diff
@@ -104,8 +111,17 @@ function ActionCreatorComponent({
         onSave={() => setShowSaveModal(true)}
         canSave={question.query().canRun()}
       />
+
       <ActionCreatorBodyContainer>
-        <QueryActionEditor question={question} setQuestion={setQuestion} />
+        <QueryActionEditor
+          question={question}
+          setQuestion={setQuestion}
+          toggleDataRef={() => setShowDataRef(prev => !prev)}
```

Comment:
fwiw, there's a [`useToggle` hook](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/hooks/use-toggle.ts) for cases like that:

```jsx
const [
  isDataRefOpen,
  { toggle: toggleDataRef, turnOff: closeDataRef }
] = useToggle(false);
```